src/install/npm.zig

```diff
@@ -824,15 +824,24 @@ pub const PackageManifest = struct {
         {
             const releases = this.pkg.releases.keys.get(this.versions);
             var i = releases.len;
+
+            var bestMatchVersion: ?Semver.Version = null;
+            var bestMatchPackage: ?PackageVersion = null;
             // For now, this is the dumb way
             while (i > 0) : (i -= 1) {
                 const version = releases[i - 1];
-                const packages = this.pkg.releases.values.get(this.package_versions);
 
-                if (group.satisfies(version)) {
-                    return .{ .version = version, .package = &packages[i - 1] };
+                // If we find one that matches, save it, but keep looping because we might find a newer match.
+                // The versions from the registry are not in any particular order.
+                if (group.satisfies(version) and (bestMatchVersion == null or Semver.Version.gt(version, bestMatchVersion.?))) {
```

Comment:
Can this use `version.order` instead of `.gt`? Something like this (but not 100% right)
```suggestion
                if (group.satisfies(version) and (bestMatchVersion == null or (version.order(best_match_version.?, string_bytes, string_bytes) == .gt)) ) {
```

Or alternatively, `gt` could be something like
```zig
pub fn gt(this: Version, rhs: Version, lhs_bytes: []const u8, rhs_bytes: []const u8) bool {
  return this.order(rhs, lhs_bytes, rhs_bytes) == .gt;
}
```

Reply:
Thanks for the quick review @Jarred-Sumner!

I started with `version.order` but my bun-foo was lacking for how to use it. I'll give it another try.